[GEOS-8940] Tile Layers page layer names bring to the "Publishing" tab by default

imranrajjad opened the pull request:
-fixed, clicking coming to layer page from Tile layers list selects 'Tile
Caching'

When clicking a layer from Tiles Admin list, the tab selected by default was Publishing when it should have been Tile Caching.

https://osgeo-org.atlassian.net/browse/GEOS-8940
## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

imranrajjad commented:
eclipse messing up the formatting as usual

imranrajjad commented on src/web/core/src/main/java/org/geoserver/web/publish/PublishedConfigurationPage.java:
```diff
@@ -1,444 +1,446 @@
+                        .indexOf(selectedTabClass);
+        if (selectedTabIndex > -1) {
+            // add differential to match index of tabPanelCustomModels with total tabs count
+            int diff = (tabbedPanel.getTabs().size() - tabPanelCustomModels.size());
```
changes of bug fix

jodygarnett commented:
If you build once on the command line it should fix the formatting for you.

imranrajjad commented:
I am on windows, thanks @jodygarnett  for the tip, will give it a shot

imranrajjad commented:
@jodygarnett merge please

jodygarnett commented on src/web/core/src/main/java/org/geoserver/web/publish/PublishedConfigurationPage.java:
```diff
@@ -1,444 +1,446 @@
-        this.inputEnabled = inputEnabled;
-        get("publishedinfo:save").setVisible(inputEnabled);
-    }
-}
```
This PR still shows every line changed, please fix line feeds. I have git client configuration instructions in the [developers guide](https://docs.geoserver.org/latest/en/developer/source.html#git-client-configuration) recently so please double check.

jodygarnett commented:
@aaime do you know if the maven source code formatting step is willing to change linefeeds?

imranrajjad commented:
looks like this PR is messed up, its better to close and create a new one, since the change is subtle.

imranrajjad commented:
@aaime this is on PR pending due to format issues, I have fixed the format issue, however the Java11 test looks like its fixed in current master, does this qualify for a merge and backport?

aaime commented:
@imranrajjad rebase then, you have to add a test for this to be merged anyways

imranrajjad commented:
@aaime sure thing (Y)

so I will

- rebase with current master
- add a wicket unit test in which user clicks layer on GWC list, which should result the Resource page showing Tile tab.

imranrajjad commented:
@aaime

rebased and added unit test, AppVeyor logs don't really describe which project is failing, but this is what I found out

```
C:\Users\appveyor\.m2\repository\org\geoserver\importer\gs-importer-web\2.17-SNAPSHOT\gs-importer-web-2.17-SNAPSHOT-test-sources.jar
[00:24:20] [INFO] 
[00:24:20] [INFO] ------------------------------------------------------------------------
[00:24:20] [INFO] Skipping GeoServer Release Module
[00:24:20] [INFO] This project has been banned from the build due to previous failures.
[00:24:20] [INFO] ------------------------------------------------------------------------
[00:24:20] [INFO] ------------------------------------------------------------------------
```

jodygarnett commented:
Thanks for the workflow improvement, as this is a functionality change it is just being applied to master.

If you wish to backport this functionality to 2.16.x we can do so after one month :)
